uqcsbot/base.py

```diff
@@ -81,6 +81,8 @@ def __init__(self, logger=None):
 
         self.channels = ChannelWrapper(self)
 
+        self.register_handler('team_join', self.channels.reload_ims)
```

Comment:
This seems inefficient and inaccurate on first glance.
1) I don't know if IMs are created upon someone joining a team, I think they're created when the person engages in an IM. Example in point: the fact that something like `im.created` and `im.open` exists. I think you should be using one of those two events-- as to which, test and find out :^)
2) We shouldn't be reloading _everyones_ IMs after just one person joins the team. Instead, add a `_handle_blah` function that individually adds the new IM channel to the channels container. likewise when `im.close` is called.

Also, move this line to underneath the other handlers so its neater :^)

I also don't know if this is the right place/way to do this, but I'll let @CameronAavik or @TRManderson work that out :^)

Reply:
Yeah I completely agree it was just from my testing that im_open didn't. im_created does though, not sure how I missed that... Makes it much easier.

It was only there because channels is declared after those lines.